src/EVM/Types.hs

```diff
@@ -432,6 +432,14 @@ data Prop where
   PBool :: Bool -> Prop
 deriving instance (Show Prop)
 
+mkPEq :: (Typeable a) => Expr a -> Expr a -> Prop
```

Comment:
Not sure what would be the best name.
I originally had `peq`, to follow the example of the helper functions for `Expr`, but I don't like that very much. I think that `mkPEq` better conveys what the function is doing. We could also use full `makePEq`.
@msooseth, opinions?

Reply:
This sounds like a good name to me.